libsolidity/formal/SSAVariable.h

```diff
@@ -0,0 +1,88 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <libsolidity/formal/SymbolicVariable.h>
+
+#include <memory>
+
+namespace dev
+{
+namespace solidity
+{
+
+class Declaration;
+
+/**
+ * This class represents the SSA representation of a program variable.
+ */
+class SSAVariable
+{
+public:
+	/// @param _decl Used to determine the type and forwarded to the symbolic var.
+	/// @param _interface Forwarded to the symbolic var such that it can give constraints to the solver.
+	explicit SSAVariable(Declaration const* _decl,
+		smt::SolverInterface& _interface);
+	SSAVariable(SSAVariable const&) = default;
+	SSAVariable(SSAVariable&&) = default;
+	SSAVariable& operator=(SSAVariable const&) = default;
+	SSAVariable& operator=(SSAVariable&&) = default;
+
+	void resetIndex();
+
+	int index() const;
```

Comment:
Please either add some documentation or rename the functions to e.g. `currentIndex` and `nextIndex`.